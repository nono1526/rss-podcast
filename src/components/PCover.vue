<template>
  <div class="flex p-5 bg-blue-400 text-white">
    <img
      class="max-w-lg w-60 border"
      :alt="imageAlt"
      :src="imageUrl"
    >
    <div class="flex-1 ml-5">
      <div>{{ author }}</div>
      <div class="text-2xl font-medium">
        {{ title }}
      </div>
      <p class="whitespace-pre-wrap mt-3 max-w">
        {{ description }}
      </p>
    </div>
  </div>
</template>

<script>
export default {
  props: {
    title: {
      type: String,
      default: ''
    },
    description: {
      type: String,
      default: ''
    },
    author: {
      type: String,
      default: ''
    },
    imageUrl: {
      type: String,
      default: ''
    },
    imageAlt: {
      type: String,
      default: ''
    }
  },
  setup (props) {
    return {
    }
  }
}
</script>

<style>

</style>
