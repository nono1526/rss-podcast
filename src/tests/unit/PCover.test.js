import PCover from '@src/components/PCover.vue'
import { shallowMount } from '@vue/test-utils'
describe('PCover.vue', () => {
  let mountFunction
  beforeEach(() => {
    mountFunction = (options = {}) => shallowMount(PCover, {
      ...options
    })
  })
  it('renders props.imgUrl when passed', () => {
    const imageUrl = 'https://test.jpg'
    const imageAlt = 'image alt'
    const wrapper = mountFunction({
      propsData: {
        imageUrl,
        imageAlt
      }
    })
    const img = wrapper.find('img')
    expect(img.attributes('src')).toBe(imageUrl)
    expect(img.attributes('alt')).toBe(imageAlt)
  })

  it('snapshot testing', () => {
    const imageUrl = 'https://test.jpg'
    const imageAlt = 'image alt'
    const wrapper = mountFunction({
      propsData: {
        imageUrl,
        imageAlt
      },
      slots: {
        default: '<div>Default slot</div>'
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
