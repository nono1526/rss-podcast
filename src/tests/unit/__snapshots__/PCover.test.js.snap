// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PCover.vue snapshot testing 1`] = `
<div
  class="flex p-5 text-white bg-gradient-to-r from-blue-500 to-green-300 "
>
  <img
    alt="image alt"
    class="max-w-lg w-60 h-60 border"
    src="https://test.jpg"
  />
  <div
    class="flex-1 ml-5"
  >
    
    <stub />
    
  </div>
</div>
`;
